lib/lambda.js

```diff
@@ -432,30 +436,61 @@ function elasticApmAwsLambda (agent) {
         agent._transport.lambdaStart()
       }
 
-      let parentId
+      const triggerType = triggerTypeFromEvent(event)
+
+      // Look for trace-context info in headers or messageAttributes.
+      let traceparent
       let tracestate
-      if (event.headers !== undefined) {
-        const normalizedHeaders = {}
-        for (const key of Object.keys(event.headers)) {
-          const value = event.headers[key]
-          const lowerCaseKey = key.toLowerCase()
-          normalizedHeaders[lowerCaseKey] = value
+      let normedHeaders
+      if (triggerType === TRIGGER_API_GATEWAY) {
+        if (event.headers) {
+          normedHeaders = lowerCaseObjectKeys(event.headers)
+        }
+      } else if (triggerType === TRIGGER_SQS_SINGLE_EVENT) {
+        const record = event.Records[0]
+        if (record.messageAttributes) {
+          normedHeaders = lowerCaseObjectKeys(headersFromSqsMessageAttributes(record.messageAttributes))
         }
-        parentId = normalizedHeaders.traceparent ? normalizedHeaders.traceparent : normalizedHeaders['elastic-apm-traceparent']
-        tracestate = normalizedHeaders.tracestate
+      } else if (triggerType === TRIGGER_SNS_SINGLE_EVENT) {
+        const record = event.Records[0]
+        if (record.Sns && record.Sns.MessageAttributes) {
+          normedHeaders = lowerCaseObjectKeys(headersFromSnsMessageAttributes(record.Sns.MessageAttributes))
+        }
+      }
```

Comment:
Not required for this PR, but I'm genuinely unsure if we should mention the agent's  automatic handling of reading the trace context headers in a Lambda environment in our message queue doc, or if that would confusing things more https://github.com/elastic/apm-agent-nodejs/blob/main/docs/message-queues.asciidoc